### PR TITLE
operator-sdk: update go dependency

### PR DESCRIPTION
**Pull request by everettraven**
operator-sdk does not support go 1.18 as of now so lock go dependency to 1.17

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Can you also add a test that will reveal an incompatible Go version next time?

**Comment by everettraven** [comment]
> Can you also add a test that will reveal an incompatible Go version next time?

@SMillerDev Tests should now be updated. Thanks!

File: Formula/operator-sdk.rb [hunk 1]
```diff
@@ -48,7 +49,14 @@ def install
       assert_match commit_regex, version_output
     end
 
-    output = shell_output("#{bin}/operator-sdk init --domain=example.com --license apache2 --owner BrewTest 2>&1", 1)
-    assert_match "failed to initialize project", output
+    mkdir "test"
+
+    chdir "test" do
```
Comment (SMillerDev) [inline Formula/operator-sdk.rb:54]: Since tests are run in a temporary directory, I'm not sure this is needed.

Comment (everettraven) [reply Formula/operator-sdk.rb:54]: If you have other suggestions for ways to work around the need for a completely empty directory I am open to implementing whatever you feel would work best. This was just the best way I could think of for a workaround

**Comment by everettraven** [comment]
Now that I see that CI is failing, it reminds me that the go@1.17 dependency does not install and set it in the PATH which is what is causing the failure in CI. Is there a way to specify that Go would need to be added to the path?

**Comment by Bo98** [comment]
> Now that I see that CI is failing, it reminds me that the go@1.17 dependency does not install and set it in the PATH which is what is causing the failure in CI. Is there a way to specify that Go would need to be added to the PATH?

You can modify the test to do that, but the best way would be a wrapper env script so that users running it don't also need to tweak that themselves.

e.g. https://github.com/Homebrew/homebrew-core/blob/e0bdda4fa89f486dabf005d425b84c9044e71eff/Formula/tfproviderlint.rb#L40-L42

**Comment by everettraven** [comment]
> > Now that I see that CI is failing, it reminds me that the go@1.17 dependency does not install and set it in the PATH which is what is causing the failure in CI. Is there a way to specify that Go would need to be added to the PATH?
> 
> 
> 
> You can modify the test to do that, but the best way would be a wrapper env script so that users running it don't also need to tweak that themselves.
> 
> 
> 
> e.g. https://github.com/Homebrew/homebrew-core/blob/e0bdda4fa89f486dabf005d425b84c9044e71eff/Formula/tfproviderlint.rb#L40-L42

Thanks for that info @Bo98 ! I will take a look at doing a wrapper env script

File: Formula/operator-sdk.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/operator-sdk.rb:57]: What is the existing file that's there? Because AFAIK that test directory is empty when you run the tests

Comment (Bo98) [reply Formula/operator-sdk.rb:57]: That'll be the hidden `.brew_home` directory.

Comment (everettraven) [reply Formula/operator-sdk.rb:57]: The exact error gotten is:
```
FATA[0000] failed to initialize project: unable to run pre-scaffold tasks of "base.go.kubebuilder.io/v3": target directory is not empty (only go.mod, go.sum, files and directories with the prefix ".", files with the suffix ".md" or capitalized files name are allowed); found existing file "Library" 
```
When injecting a test that runs and checks the output of an `ls -la` command, it showed the following as the output of the `ls -la`:

```
total 4
drwx------ 1 linuxbrew linuxbrew  30 May  3 18:36 .
drwxrwxrwt 1 root      root      138 May  3 18:36 ..
-rw-r--r-- 1 linuxbrew linuxbrew  79 May  3 18:36 .bazelrc
drwxr-xr-x 1 linuxbrew linuxbrew  12 May  3 18:36 Library
```

Comment (everettraven) [reply Formula/operator-sdk.rb:57]: Hidden directories are allowed by operator-sdk but in this case it seems like the `Library` directory is causing an issue.

Comment (Bo98) [reply Formula/operator-sdk.rb:57]: Ah forgot `.brew_home` is there in build environments but not test environments.

**Review by SMillerDev — DISMISSED** [review]

**Comment by cho-m** [comment]
You will need to fix commit history to get rid of `automerge-skip` label. Specifically, `Merge branch 'Homebrew:master' ...` commits cannot be autosquashed by bot. To avoid this in future, use git's rebase rather than merge (and also avoid GitHub's web options to fetch upstream)

Only commits directly related to `operator-sdk.rb` should be included in commit history, preferably squashed down to a single commit.

File: Formula/operator-sdk.rb [hunk 2]
```diff
@@ -21,23 +21,27 @@ class OperatorSdk < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "b951111c79c494c2e3a4ea06562306d698959a6807b002132c578cca64ad75eb"
   end
 
-  depends_on "go"
+  # TODO(everettraven): update this to be "go" when operator-sdk add support for go 1.18
```
Comment (SMillerDev) [inline Formula/operator-sdk.rb:24]: This should reference an upstream issue since the TODO won't be seen

File: Formula/operator-sdk.rb [hunk 3]
```diff
@@ -48,7 +52,14 @@ def install
       assert_match commit_regex, version_output
     end
 
-    output = shell_output("#{bin}/operator-sdk init --domain=example.com --license apache2 --owner BrewTest 2>&1", 1)
-    assert_match "failed to initialize project", output
+    mkdir "test"
+
+    chdir "test" do
```
Comment (SMillerDev) [inline Formula/operator-sdk.rb:57]:
```suggestion
    mkdir "test" do
```

Comment (everettraven) [reply Formula/operator-sdk.rb]: updated in commit https://github.com/Homebrew/homebrew-core/pull/100353/commits/5a4283ab2399d369e2d3b942c946d4aa348e099a

File: Formula/operator-sdk.rb (hunk 2 above)
Comment (everettraven) [reply Formula/operator-sdk.rb]: I changed this in commit https://github.com/Homebrew/homebrew-core/pull/100353/commits/5a4283ab2399d369e2d3b942c946d4aa348e099a

Let me know if it needs any further changes

File: Formula/operator-sdk.rb [hunk 4]
```diff
@@ -13,31 +13,36 @@ class OperatorSdk < Formula
   end
 
   bottle do
```
Comment (SMillerDev) [inline Formula/operator-sdk.rb:15]: This block should not be modified

Comment (everettraven) [reply Formula/operator-sdk.rb]: Should be fixed as of commit https://github.com/Homebrew/homebrew-core/pull/100353/commits/9d2700a770552f456b1448b1eadc6ba43db10769

**Review by SMillerDev — DISMISSED** [review]

**Comment by SMillerDev** [comment]
Can you rebase the branch and squash all commits?

**Comment by everettraven** [comment]
I am having some issues with getting a rebase to work out in my favor (i'm not to familiar with using `git rebase`). I think for simplicity sake I am going to close this PR and create a new one that has one commit that changes exactly what needs to be changed.

**Pull request closed without merging**